### PR TITLE
Add redundant typed throws rule

### PR DESCRIPTION
**Pull request by miguel-jimenez-0529**
This PR adds a new `redundantTypedThrows` rule, which removes a typed throws (introduced in Swift 6.0) if it is redundant. 

For example:

```diff
- func foo() throws(Never) -> Int {
+ func foo() -> Int {
      return 0
  }

- func foo() throws(any Error) -> Int {
+ func foo() throws -> Int {
      throw MyError.foo
  }
```
We don't remove `(Error)` because we can't guarantee it will reference the `Swift.Error` protocol
(it's relatively common to define a custom error like `enum Error: Swift.Error { ... }`).

### CODE REVIEW

**Comment by miguel-jimenez-0529** [comment]
cc: @calda

File: Tests/RulesTests+Redundancy.swift [hunk 1]
```diff
@@ -10110,4 +10110,65 @@ class RedundancyTests: RulesTests {
...
+            throw .foo
+        }
+
+        func foo() throws(Error) -> Int {
+            throw MyError.foo
+        }
+        """
+
+        let options = FormatOptions(swiftVersion: "6.0")
+        testFormatting(for: input, output, rule: FormatRules.redundantTypedThrows, options: options)
```
Comment (calda) [inline Tests/RulesTests+Redundancy.swift:10172]: Looks like the `output` here is redundant?

```suggestion

        let options = FormatOptions(swiftVersion: "6.0")
        testFormatting(for: input, rule: FormatRules.redundantTypedThrows, options: options)
```

**Review by calda — APPROVED** [review]
Looks great! Nice work @miguel-jimenez-0529!

File: Tests/RulesTests+Redundancy.swift (hunk 1 above)
Comment (miguel-jimenez-0529) [reply Tests/RulesTests+Redundancy.swift]: Yeah just noticed that. Should addressed in the last commit

**Pull request merged**